### PR TITLE
Search components migration

### LINKED ISSUE #615: It is not possible to modify the start date of a dataset outside of the calendar widget
Steps to reproduce the bug:
- I edit a dataset
- I set the start date to date_1 **with the calendar widget**
- I save the dataset
- I edit the dataset
- I set the start date  to date_2 **without the calendar widget**
- I save the dataset
- I edit the dataset

Expected behavior:
- The start date should be date_2

Actual behaviour:
- The start date is date_1

### PR DESCRIPTION
**Pull request by noirbizarre**
This PR is part of #526:
- Pure Vue.js search facets.
- Drop some unused dependencies (sliders)
- Use the right bundle in search/listing views
- Improve and fix the datepicker:
    - drop bootstrap-datepicker:
      - lighter style
      - flexbox+button based instead of the hackish table based one
    - Proper sizing and positionning in dropdowns
    - Fix initial value not being displayed
    - Make it usable on keyboard
    - Allows to define `min` and `max` values to disable some dates
    - Keyboard input is reflected into the calendar (fix #615 )

### CODE REVIEW

File: js/components/facets/temporal-coverage.vue [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    methods: {
+        apply() {
+            const filter = `${this.currentMin.format(ISO_FORMAT)}-${this.currentMax.format(ISO_FORMAT)}`;
+            const search = new URLSearchParams(window.location.search);
+            search.set(this.name, filter);
+            window.location.search = search.toString();
+        },
+        onFocus(e) {
+            if (!this.picking || e.target !== this.pickedField) {
+                console.log('should focus');
```
Comment (davidbgk) [inline js/components/facets/temporal-coverage.vue]: Remove?

File: js/components/facets/temporal-coverage.vue [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                this.$nextTick(this.$refs.calendar.focus);
+            }
+            this.picking = true;
+            this.pickedField = e.target;
+        },
+        onChange(e) {
+            try {
+                this.currentValue = moment(e.target.value, DATE_FORMAT);
+            } catch(e) {}
```
Comment (davidbgk) [inline js/components/facets/temporal-coverage.vue]: Notify somehow?

File: js/components/form/date-picker.vue [hunk 3]
```diff
@@ -49,31 +49,36 @@ export default {
...
     },
     methods: {
         onFocus() {
+            if (!this.picking) this.$nextTick(this.$refs.calendar.focus);
             this.picking = true;
         },
+        onChange(e) {
+            try {
+                this.value = moment(e.target.value, this.dateFormat);
+            } catch(e) {}
```
Comment (davidbgk) [inline js/components/form/date-picker.vue]: Same here.

File: js/components/form/daterange-picker.vue [hunk 4]
```diff
@@ -108,15 +128,26 @@ export default {
...
             this.picking = true;
             this.pickedField = e.target;
-            this.hiddenField = e.target == this.$els.startInput
-                ? this.$els.startHidden
-                : this.$els.endHidden;
+        },
+        onChange(e) {
+            try {
+                this.currentValue = moment(e.target.value, this.dateFormat);
+            } catch(e) {}
```
Comment (davidbgk) [inline js/components/form/daterange-picker.vue]: And here :)

File: js/front/search.js [hunk 5]
```diff
@@ -1,44 +1,83 @@
...
- * Generic site display page JS module
+ * Search display page JS module
  */
 import FrontMixin from 'front/mixin';
 
 import log from 'logger';
 import Vue from 'vue';
+import velocity from 'velocity-animate';
 
-// Legacy depdencies soon to be dropped
```
Comment (davidbgk) [inline js/front/search.js]: 🎉

File: udata/templates/search.html [hunk 6]
```diff
@@ -21,39 +21,39 @@
...
+        <li{% if not datasets and not reuses and not organizations %} class="active"{% endif %} data-tab="territories">
             {% set total=territories|length %}
             {% if total==1 %}
                 <a href="{{ territories.0.url }}">
                     {{ _('Territory:') }}
                     <span class="badge">{{ territories.0.name }}</span>
                 </a>
             {% else %}
-                <a href="#territories" data-toggle="tab">
+                <a href@click.prevent="setTab('territories')">
```
Comment (davidbgk) [inline udata/templates/search.html]: Missing space before `@`?

File: js/components/calendar.vue [hunk 7]
```diff
@@ -82,59 +49,74 @@ import moment from 'moment';
...
+        months() {
             return moment.monthsShort();
         },
-        monthDisplay: function() {
-            return moment()
-                .month(this.currentMonth)
-                .year(this.currentYear)
-                .format('MMMM YYYY');
+        monthDisplay() {
+            return moment().month(this.currentMonth).year(this.currentYear).format('MMMM YYYY');
```
Comment (davidbgk) [inline js/components/calendar.vue]: Turn this format into a constant?

File: js/components/calendar.vue [hunk 8]
```diff
@@ -145,76 +127,289 @@ export default {
...
+    }
+
+	button {
+	    height: @unit;
+        width: 100%;
+        background: none;
+        border: none;
+        border-radius: @border-radius-base;
+
+        &:hover {
```
Comment (davidbgk) [inline js/components/calendar.vue]: Weird indent here and below

File: js/components/facets/temporal-coverage.vue (hunk 1 above)
Comment (noirbizarre) [inline js/components/facets/temporal-coverage.vue]: :+1:

File: js/front/search.js (hunk 5 above)
Comment (noirbizarre) [inline js/front/search.js]: ;)

File: js/components/calendar.vue (hunk 8 above)
Comment (noirbizarre) [inline js/components/calendar.vue]: Fixed: was mix of spaces and tabs (copy-paste :/). Now it's space everywhere

File: js/components/facets/temporal-coverage.vue [hunk 9]
```diff
@@ -34,6 +34,7 @@
 <script>
 import Calendar from 'components/calendar.vue';
 import moment from 'moment';
+import log from 'logger';
```
Comment (davidbgk) [inline js/components/facets/temporal-coverage.vue]: I can't see where `log` is finally in use, 👍 for comments

File: js/components/calendar.vue [hunk 10]
```diff
@@ -48,6 +48,7 @@
 import moment from 'moment';
 
 const VIEWS = ['days', 'months', 'years'];
+const MONTH_FORMAT = 'MMMM YYYY';
```
Comment (davidbgk) [inline js/components/calendar.vue]: Maybe we should put all date-related constants somewhere?

Comment (noirbizarre) [inline js/components/calendar.vue]: Yes, but in this this case this very specific to the way its used/displayed into the widget

File: js/components/facets/temporal-coverage.vue (hunk 9 above)
Comment (noirbizarre) [inline js/components/facets/temporal-coverage.vue]: Oups, forgot to remove (I started with logging first and this how I remember why the catch blocks where empty)

**Pull request merged**